ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -1,32 +1,137 @@
-import {cdrVersionStore, currentWorkspaceStore} from 'app/utils/navigation';
+import {cdrVersionStore, currentWorkspaceStore, navigate, routeConfigDataStore} from 'app/utils/navigation';
 import {mount} from 'enzyme';
-import {CdrVersionsApi, WorkspaceAccessLevel} from 'generated/fetch';
+import {WorkspaceAccessLevel, WorkspacesApi, SpecificPopulationEnum} from 'generated/fetch';
 import * as React from 'react';
-import {ClusterApiStub} from 'testing/stubs/cluster-api-stub';
-import {WorkspaceEdit, WorkspaceEditMode} from './workspace-edit';
-import {workspaceStubs} from 'testing/stubs/workspaces-api-stub';
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {WorkspaceEdit, WorkspaceEditMode, WorkspaceEditProps} from './workspace-edit';
+import {workspaceStubs, WorkspacesApiStub} from 'testing/stubs/workspaces-api-stub';
 import {cdrVersionListResponse} from 'testing/stubs/cdr-versions-api-stub';
 
+jest.mock('app/utils/navigation', () => ({
+  ...(require.requireActual('app/utils/navigation')),
+  navigate: jest.fn()
+}));
 
-let props: {};
-const workspace = {
-  ...workspaceStubs[0],
-  accessLevel: WorkspaceAccessLevel.OWNER,
-};
+describe('WorkspaceEdit', () => {
+  let workspacesApi: WorkspacesApiStub;
+  const workspace = {
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.OWNER,
+  };
 
-const component = () => {
-  return mount(<WorkspaceEdit {...props}/>);
-};
+  const component = () => {
+    return mount(<WorkspaceEdit cancel={() => {}} />);
+  };
 
-beforeEach(() => {
-  registerApiClient(CdrVersionsApi, new ClusterApiStub());
-  props = {workspace: workspace, routeConfigData: {mode: WorkspaceEditMode.Create}};
-  currentWorkspaceStore.next(workspace);
-  cdrVersionStore.next(cdrVersionListResponse);
-});
+  beforeEach(() => {
+    workspace.researchPurpose = {
+      ...workspace.researchPurpose,
+      intendedStudy: 'greyscale',
+      anticipatedFindings: 'everything',
+      reasonForAllOfUs: 'science',
+      drugDevelopment: true
+    };
+
+    workspacesApi = new WorkspacesApiStub([workspace]);
+    registerApiClient(WorkspacesApi, workspacesApi);
+
+    currentWorkspaceStore.next(workspace);
+    cdrVersionStore.next(cdrVersionListResponse);
+    routeConfigDataStore.next({mode: WorkspaceEditMode.Create});
+  });
+
+  it('displays workspaces create page', async () => {
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
+  });
+
+  it('displays workspaces duplicate page', async () => {
+    routeConfigDataStore.next({mode: WorkspaceEditMode.Duplicate});
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
+  });
+
+  it('displays workspaces edit page', async () => {
+    routeConfigDataStore.next({mode: WorkspaceEditMode.Edit});
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
```

Comment:
since we're testing three different rendering of this page, can we assert something that verified that the right page was rendered?

Anything small and unique like a word in the title.

Reply:
Thanks, meant to come back to this but I forgot - done.